File: src/main/java/cal/fstg/java/service/AdminService.java
//package cal.fstg.java.service;
//
//import cal.fstg.java.dao.AdminRepository;
//import cal.fstg.java.dao.DocumentRepository;
//import cal.fstg.java.model.Admin;
//import cal.fstg.java.model.Document;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.stereotype.Service;
//
//import java.util.List;
//import java.util.Optional;
//
//@Service
//public class AdminService {
//
//    private final AdminRepository adminRepository;
//
//    @Autowired
//    public AdminService(AdminRepository adminRepository) {
//        this.adminRepository = adminRepository;
//    }
//
//    public List<Admin> getAll() {
//        return adminRepository.findAll();
//    }
//
//    public Optional<Admin> getOne(long id) {
//        return adminRepository.findById(id);
//    }
//
//    public Admin save(Admin admin) {
//        return adminRepository.save(admin);
//    }
//
//    public void delete(long id) {
//        adminRepository.deleteById(id);
//    }
//
//}

File: src/main/kotlin/cal/fstg/stagerkotlin/service/AdminService.kt
package cal.fstg.stagerkotlin.service

import cal.fstg.stagerkotlin.dao.AdminRepository
import cal.fstg.stagerkotlin.model.Admin
import cal.fstg.stagerkotlin.model.User
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.reactor.flux
import kotlinx.coroutines.reactor.mono
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class AdminService(@Autowired private val adminRepository: AdminRepository) {

    @ExperimentalCoroutinesApi
    fun getAll() = flux<Admin> { adminRepository.findAll() }

    fun getOne(id: String) = mono { adminRepository.findById(id) }

    fun save(admin: Admin) = mono { adminRepository.save(admin) }

    fun delete(id: String) = adminRepository.deleteById(id)
}